ledger/src/blockstore.rs

```diff
@@ -1117,6 +1111,70 @@ impl Blockstore {
             &mut write_batch,
         )?;
 
+        for (erasure_set, working_erasure_meta) in erasure_metas.iter() {
+            if !working_erasure_meta.should_write() {
+                // Not a new erasure meta
+                continue;
+            }
+            let (slot, _) = erasure_set.store_key();
+            // First coding shred from this erasure batch, check the forward merkle root chaining
+            if !self.has_duplicate_shreds_in_slot(slot) {
+                let erasure_meta = working_erasure_meta.as_ref();
+                let shred_id = ShredId::new(
+                    slot,
+                    erasure_meta
+                        .first_received_coding_shred_index()
+                        .expect("First received coding index must exist for all erasure metas"),
+                    ShredType::Code,
+                );
+                let shred = Shred::new_from_serialized_shred(
```

Comment:
in a future pr i can remove this deserialize and the one below since we serialize this anyway in window service in order to push to gossip.